src/main/java/Duke/todoList/todoList.java

```diff
@@ -0,0 +1,110 @@
+        }
+        for(int i = 0; i < totalTasks; i++){
+            if(i < totalTasks - 1) {
+                output += Integer.toString(i + 1)
+                        + ". " + taskList.get(i).displayType()
+                        + " " + taskList.get(i).displayResolved()
+                        + " " + taskList.get(i).displayDescription()
+                        + " " + taskList.get(i).displayDate() + "\n";
+            }else{
+                output += Integer.toString(i + 1)
+                        + ". " + taskList.get(i).displayType()
+                        + " " + taskList.get(i).displayResolved()
+                        + " " + taskList.get(i).displayDescription()
+                        + " " + taskList.get(i).displayDate();
+            }
+        }
+        return output;
+    }
+
+    public String resolveTask(String num){ // resolves task and returns confirmation if task is not resolved
+        Integer taskNum = Integer.parseInt(num) - 1;
+        if (taskList.get(taskNum).isResolved()){
+            return "Task is already done!";
+        }
+        taskList.get(taskNum).resolve();
+        this.leftoverTasks--;
+        return "Great! Task ["+ taskList.get(taskNum).displayDescription() + "] has been marked as done!"
+                + "\nYou now have " + this.tasksLeft() + " task(s) left undone in the list!";
+
+    }
+
+    public int tasksLeft(){ // returns number of undone tasks
+        return this.leftoverTasks;
+    }
+
+    public int tasksTotal(){ // returns total tasks in list
+        return this.totalTasks;
+    }
+
+    public String deleteTask(String num){ // delete the specified task
+        Integer taskNum = Integer.parseInt(num) - 1;
+        improvedTask temp = taskList.get(taskNum);
+        taskList.remove(temp);
+        totalTasks--;
+        if(!temp.isResolved()) {
+            leftoverTasks--;
+        }
+        String out = "Noted. I've removed this task: \n     "
+                   + temp.displayType()
+                   + " " + temp.displayResolved()
+                   + " " + temp.displayDescription()
+                   + " " + temp.displayDate() + "\n"
+                   + "Now you have " + totalTasks + " task(s) total left in the list!";
+        return out;
+    }
+
+    public String tasksUpdate(){ // returns current lists of tasks that can be parsed by updateFile method
+        String out = "";
+        for (int i = 0; i < totalTasks; i++){
+            if (taskList.get(i).displayType().equals("[T]")) {
+                out += taskList.get(i).displayType().substring(1, 2) + "/"
+                        + taskList.get(i).displayResolved().substring(1, 2) + "/"
+                        + taskList.get(i).displayDescription() + "/\n";
+            }else{
+                out += taskList.get(i).displayType().substring(1, 2) + "/"
+                        + taskList.get(i).displayResolved().substring(1, 2) + "/"
+                        + taskList.get(i).displayDescription() + "/"
```

Comment:
One of the quality of good code is keeping things [simple](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#not-too-long-not-too-short). You could try to maintain this principle for the code block here.